File: minipack-core/src/test/java/org/minipack/core/example/ReadFromStream.java
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.minipack.core.example;

import org.junit.jupiter.api.Test;
import org.minipack.core.MessageReader;

import java.io.IOException;
import java.io.InputStream;

@SuppressWarnings("unused")
class ReadFromStream extends Example {
  // -8<- [start:snippet]
  void read(InputStream stream) throws IOException {
    try (var reader = MessageReader.of(stream)) {
      var string = reader.readString();
      var number = reader.readInt();
    }
  }
  // -8<- [end:snippet]

  @Test
  void test() throws IOException {
    writer.write("Hello, MiniPack!");
    writer.write(42);
    writer.close();
    read(inStream);
  }
}

File: minipack-kotlin/src/test/kotlin/org/minipack/kotlin/example/ReadFromStream.kt
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package org.minipack.kotlin.example

import org.junit.jupiter.api.Test
import java.io.InputStream
import org.minipack.kotlin.MessageReaders

@Suppress("UNUSED_VARIABLE")
class ReadFromStream : Example() {
  // -8<- [start:snippet]
  fun read(stream: InputStream) {
    MessageReaders.of(stream).use { reader ->
      val string = reader.readString()
      val number = reader.readInt()
    }
  }
  // -8<- [end:snippet]

  @Test
  fun test() {
    writer.write("Hello, MiniPack!")
    writer.write(42)
    writer.close()
    read(inStream)
  }
}
